### ISSUE TITLE
Out-of-date documentation for binding system

### ISSUE THREAD

**Issue by qstrahl** [fish-shell/fish-shell#6075]
This appears to be out of date; https://fishshell.com/docs/current/index.html#editor

As I understand it, the new method for assigning custom key bindings is to define a `fish_user_key_bindings` function, not to set `fish_key_bindings` to your own private function.

Not that it's super relevant, but... versions:
```
fish, version 3.0.2
3.0.2
Linux vergadain 5.2.9-arch1-1-ARCH #1 SMP PREEMPT Fri Aug 16 11:29:43 UTC 2019 x86_64 GNU/Linux
tmux-256color
```

**Comment by zanchey** [comment]
I'm not actually entirely clear what the current recommendation is - @faho?

Now that you can put `bind` entries straight into `config.fish`, there's less need for `fish_user_key_bindings`, although I suspect if you are changing your bindings it would be better to use it.

**Comment by qstrahl** [comment]
Hmm... that does seem to be the most recent (and cleanest) standard. Docs were even further behind than I thought.

**Comment by faho** [comment]
Okay, so the current recommendation is:

- If you just want to set a couple of keys (e.g. set ctrl-g to launch tetris), put `bind` statements somewhere. config.fish is fine, `fish_user_key_bindings` is also available if you wish to organize them.

- If you want to change the entire binding paradigm, e.g. if you wanted to make fish behave like kakoune or if you want to use vi-mode, set $fish_key_bindings.

Now, the hybrid binding thing is the latter. It changes the binding paradigm to use both vi and emacs-style keys. However, the main reason for this to use a new function is because the binding functions of course erase the preset bindings so you get a clean slate when switching, so you *have* to use a hybrid function. So the documentation is not out of date, though the solution isn't as great as it could be.

**Comment by zanchey** [comment]
Fixed in 82de51b9d3.
